File: trunk/JAVA/jzbus/src/test/java/net/zbus/loadbalance/service/PooledService.java
package net.zbus.loadbalance.service;

import net.zbus.ServiceHandler;
import net.zbus.Worker;
import net.zbus.WorkerPool;
import net.zbus.WorkerPoolConfig;
import net.zbus.ZMsg;

public class PooledService {
 
	public static void main(String[] args) {
		//1) 配置服务信息
		WorkerPoolConfig cfg = new WorkerPoolConfig();
		cfg.setService("PooledService1");
		cfg.setMode(Worker.MODE_LB);
		cfg.setBrokers(new String[]{"127.0.0.1:15555","42.120.17.176:15555"}); //总线地址（多总线注册）
		
		int threadCount = 2;
		WorkerPool wc = new WorkerPool(cfg); 
		System.out.format("Pooled Workers(%d) Run...\n", threadCount);
		
		//2)以指定的线程数运行服务处理业务逻辑代码
		wc.run(threadCount, new ServiceHandler() { 			
			@Override
			public ZMsg handleRequest(ZMsg request) {
				request.dump(); 
				ZMsg result = new ZMsg();
				result.pushBack("this is from pooled java worker");
				return result;
			} 
		});
	}
}

File: MS.NET/clrzbus/examples/loadbalance/service/PooledService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class MyServiceHandler : ServiceHandler
    {
        public ZMsg HandleRequest(ZMsg request)
        {
            request.Dump();

            ZMsg result = new ZMsg();
            result.PushBack("this is from pooled c# worker");

            return result;
        }
    }

    class PooledService
    {
        public static void Main(string[] args)
        {
            //1) 配置服务信息
            WorkerPoolConfig config = new WorkerPoolConfig();
            config.Service = "MyService";
            config.Mode = WorkerConfig.MODE_LB;

            
            int threadCount = 2;
            Console.WriteLine("Pooled Workers({0}) Run...", threadCount);
            WorkerPool pool = new WorkerPool(config);
            //2)以指定的线程数运行服务处理业务逻辑代码
            pool.Run(threadCount, new MyServiceHandler()); 
             
        } 
    }
}
